src/Components/News/NewsCreateEdit.js

```diff
@@ -0,0 +1,231 @@
+import { useParams } from 'react-router';
+import '../../Components/FormStyles.css';
+import { CKEditor } from '@ckeditor/ckeditor5-react';
+import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
+import {useDropzone} from 'react-dropzone';
+import Box from '@mui/material/Box';
+import TextField from '@mui/material/TextField';
+
+
+const editorConfiguration = {
+    toolbar: [
+        'heading',
+        '|',
+        'bold',
+        'italic',
+        'link',
+        'bulletedList',
+        'numberedList',
+        'blockQuote',
+        'undo',
+        'redo'
+    ]
+};
+
+const datosForm = {
+    category: '',
+    title: '',
+    content: '',
+    image: '',
+}
+//dropzone--------------------------------
+const thumbsContainer = {
+    display: 'flex',
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    marginTop: 16
+  };
+  
+  const thumb = {
+    display: 'inline-flex',
+    borderRadius: 2,
+    border: '1px solid #eaeaea',
+    marginBottom: 8,
+    marginRight: 8,
+    width: 100,
+    height: 100,
+    padding: 4,
+    boxSizing: 'border-box'
+  };
+  
+  const thumbInner = {
+    display: 'flex',
+    minWidth: 0,
+    overflow: 'hidden'
+  };
+  
+  const img = {
+    display: 'block',
+    width: 'auto',
+    height: '100%'
+  };
+
+function NewsCreateEdit() {
+    
+    const { id } = useParams() //Esto valida si se esta editando o creando un post mediante paso de parametros
+    //const refP = useRef() //cambia la visibilidad del <p> "insertar imagen"
+    const urlCategoria = 'http://ongapi.alkemy.org/public/api/categories';
+    const urlEditarNovedad = `http://ongapi.alkemy.org/public/api/news/${id}`
+    const urlCrearNovedad = 'http://ongapi.alkemy.org/public/api/news'
+    const [categorias, setcategorias] = useState([]) //obtiene las categorias del endpoint
+    const [initialValues, setInitialValues] = useState(datosForm);
+    const [files, setFiles] = useState([]);
+    const {getRootProps, getInputProps} = useDropzone({
+      accept: 'image/png, image/jpg',
+      onDrop: acceptedFiles => {
+        setInitialValues({...initialValues, image: (acceptedFiles.map(file => Object.assign(file, {
+            preview: URL.createObjectURL(file)
+        })))})
+        setFiles(acceptedFiles.map(file => Object.assign(file, {
+          preview: URL.createObjectURL(file)
+        })));
+      }
+    });
+    
+    const thumbs = files.map(file => (
+      <div style={thumb} key={file.name}>
+        <div style={thumbInner}>
+          <img
+            src={file.preview}
+            style={img}
+            alt='img not found'
+          />
+        </div>
+      </div>
+    ));
+  
+    useEffect(() => () => {
+      // Make sure to revoke the data uris to avoid memory leaks
+      files.forEach(file => URL.revokeObjectURL(file.preview));
+    }, [files]);
+
+    const handleChange = (e) => { //actualiza el estado de title y category
+        if (e.target.name === 'title') {
```

Comment:
instead of an if use a switch, is faster and if you need to add more attributes in the future is clearer and more simple